components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -7225,16 +7281,38 @@ public static JSONObject getTenantDefaultRoles(String tenantDomain) throws APIMa
     }
 
     /**
-     * @param config JSON configuration object with scopes and associated roles
+     * @param scopesConfig JSON configuration object with scopes and associated roles
+     * @param roleMappings JSON Configuration object with role mappings
      * @return Map of scopes which contains scope names and associated role list
      */
-    public static Map<String, String> getRESTAPIScopesFromConfig(JSONObject config) {
+    public static Map<String, String> getRESTAPIScopesFromConfig(JSONObject scopesConfig, JSONObject roleMappings) {
         Map<String, String> scopes = new HashMap<String, String>();
-        JSONArray scopesArray = (JSONArray) config.get("Scope");
+        JSONArray scopesArray = (JSONArray) scopesConfig.get("Scope");
         for (Object scopeObj : scopesArray) {
             JSONObject scope = (JSONObject) scopeObj;
             String scopeName = scope.get(APIConstants.REST_API_SCOPE_NAME).toString();
             String scopeRoles = scope.get(APIConstants.REST_API_SCOPE_ROLE).toString();
+            if (roleMappings != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("REST API scope role mappings exist. Hence proceeding to swap original scope roles "
+                            + "for mapped scope roles.");
+                }
+                //split role list string read using comma separator
+                List<String> originalRoles = Arrays.asList(scopeRoles.split("\\s*,\\s*"));
+                List<String> mappedRoles = new ArrayList<String>();
+                for (String role : originalRoles) {
+                    String mappedRole = (String) roleMappings.get(role);
+                    if (mappedRole != null) {
+                        if (log.isDebugEnabled()) {
+                            log.debug(role + " was mapped to " + mappedRole);
+                        }
+                        mappedRoles.add(mappedRole);
+                    } else {
+                        mappedRoles.add(role);
+                    }
+                }
+                scopeRoles = mappedRoles.toString().replace("[", "").replace("]", "").replace(" ", "");
```

Comment:
Are you trying to do below?
scopeRoles = String.join(",", mappedRoles);

Reply:
yes, fixed with https://github.com/wso2/carbon-apimgt/pull/8164/commits/1bffd034ecf46cff97a7ff824b6b9c71b0754698